packages/pages/src/common/src/template/internal/loader.ts

```diff
@@ -0,0 +1,73 @@
+import { TemplateModule } from "../types.js";
+import {
+  convertTemplateModuleToTemplateModuleInternal,
+  TemplateModuleInternal,
+} from "./types.js";
+import esbuild from "esbuild";
+import { importFromString } from "module-from-string";
+
+const TEMP_DIR = ".temp";
+
+/**
+ * Loads all templates in the project.
+ * @param templateModulePaths the templates filepaths to load as modules
+ * @param transpile set to true if the templates need to be transpiled (such as when they are in tsx format)
+ * @param adjustForFingerprintedAsset removes the fingerprint portion (for server bundles)
+ * @returns Promise<{@link TemplateModuleCollection}>
+ */
+export const loadTemplateModules = async (
+  templateModulePaths: string[],
+  transpile: boolean,
```

Comment:
when would we not want to transpile or adjust for fingerprinted assets? 

I'd think we'd *always* want to do this.

Reply:
This function is used by the generate features.json command, which loads the raw tsx file, so in that case we need to transpile. But this is also used by the plugin code (was trying to consolidate the loading of templates code), but at that point it's using the server assets which are already transpiled by this point. The server bundles also need to do the adjusting thing.

Do you think we shouldn't use in both places or do something else?

Reply:
I see. Sounds good then.